src/ucp/wireup/wireup_cm.c

```diff
@@ -572,6 +572,15 @@ static unsigned ucp_cm_client_connect_progress(void *arg)
               ucp_ep->flags, ucp_ep->cfg_index);
     ucs_log_indent(1);
 
+    if (!(ucp_ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED)) {
+        /* If LOCAL_CONNECTED flag was removed from the endpoint, it means that
+         * the endpoint failed and CM was disconnected from a peer. So, no need
+         * for futher actions */
+        ucs_assert(ucp_ep->flags & UCP_EP_FLAG_FAILED);
+        status = UCS_OK;
+        goto out;
```

Comment:
sorry, but looking at this again, maybe better to
1. remove notify/client_connect progress callbacks before calling cm_disconnect (**instead** of during ep destroy)
2. assume that CM will also not generate async events after disconnect is called
2. in here, assert that LOCAL_CONNECTED flag is set
WDYT?

Reply:
done